### ISSUE TITLE
error: foreach() argument must be of type array|object, bool given

### ISSUE THREAD

**Issue by lazka** [dragonmantank/cron-expression#134]
Starting with a recent update the following fails:

```php
$cron = new CronExpression('0 0 1 1 *');
$cron->getPreviousRunDate(new \DateTimeImmutable('2021-09-07T09:36:00Z'));
```
```
foreach() argument must be of type array|object, bool given

vendor/dragonmantank/cron-expression/src/Cron/HoursField.php:100
vendor/dragonmantank/cron-expression/src/Cron/HoursField.php:52
vendor/dragonmantank/cron-expression/src/Cron/CronExpression.php:515
vendor/dragonmantank/cron-expression/src/Cron/CronExpression.php:298
```

It worked with v3.2.0 and fails with v3.2.1. Tested with PHP 7.3-8.1

**Comment by dragonmantank** [comment]
This should be fixed in v3.3.1, which is going out in just a few minutes. Thanks!

**Comment by lazka** [comment]
Thank you! 🙏
